tests/test_managed_allow_list.py

```diff
@@ -1,160 +1,95 @@
+    return proxy_contract
 
 
-@pytest.fixture()
-def fee_model(project, deployer, coordinator, fee_token):
-    contract = project.FlatRateFeeModel.deploy(
-        coordinator.address, fee_token.address, FEE_RATE, sender=deployer
-    )
-    return contract
+def test_authorize_using_global_allow_list(coordinator, deployer, initiator, managed_allow_list):
+    # This block mocks the signature of a threshold decryption request
+    w3 = Web3()
+    data = os.urandom(32)
+    digest = Web3.keccak(data)
+    signable_message = encode_defunct(digest)
+    signed_digest = w3.eth.account.sign_message(signable_message, private_key=deployer.private_key)
+    signature = signed_digest.signature
 
+    ritual_id = 0
+    cohort_admin_role = managed_allow_list.cohortAdminRole(ritual_id)
+    auth_admin_role = managed_allow_list.authAdminRole(ritual_id)
 
-@pytest.fixture()
-def brand_new_managed_allow_list(project, coordinator, subscription, fee_model, authority):
-    return project.ManagedAllowList.deploy(
-        coordinator.address, subscription.address, sender=authority
-    )
+    # Not authorized
+    assert not managed_allow_list.isAuthorized(0, bytes(signature), bytes(digest))
 
+    # Negative test cases for authorization
+    with ape.reverts("Only auth admin is permitted"):
+        managed_allow_list.authorize(ritual_id, [deployer.address], sender=deployer)
 
-@pytest.fixture()
-def managed_allow_list(brand_new_managed_allow_list, coordinator, deployer, authority):
-    coordinator.mockNewRitual(authority, sender=deployer)
-    return brand_new_managed_allow_list
+    with ape.reverts("Ritual is not initiated"):
+        managed_allow_list.initializeCohortAdminRole(ritual_id, sender=deployer)
 
+    coordinator.initiateRitual(ritual_id, initiator, sender=initiator)
 
-def test_initial_parameters(managed_allow_list, coordinator, admin, encryptor):
-    assert managed_allow_list.coordinator() == coordinator.address
-    assert not managed_allow_list.getAllowance(RITUAL_ID, admin.address)
-    assert not managed_allow_list.authActions(RITUAL_ID)
+    managed_allow_list.initializeCohortAdminRole(ritual_id, sender=initiator)
+    assert managed_allow_list.hasRole(cohort_admin_role, initiator)
 
+    managed_allow_list.grantRole(auth_admin_role, deployer, sender=initiator)
+    assert managed_allow_list.hasRole(auth_admin_role, deployer)
+    managed_allow_list.authorize(ritual_id, [deployer.address], sender=deployer)
 
-def test_add_administrators(managed_allow_list, authority, admin):
-    # Only authority can add administrators
-    with ape.reverts("Only cohort authority is permitted"):
-        managed_allow_list.addAdministrators(RITUAL_ID, [admin.address], ADMIN_CAP, sender=admin)
+    managed_allow_list.grantRole(auth_admin_role, initiator, sender=initiator)
+    with ape.reverts("Encryptor must be authorized by the sender first"):
+        managed_allow_list.deauthorize(ritual_id, [deployer.address], sender=initiator)
```

Comment:
Here's some food for thought for a new test:
* Cohort admin authorizes auth admin A
* Cohort admin authorizes auth admin B
* A authorizes encryptor E
* B authorizes encryptor E
* A deauthorizes encryptor E

What should happen with E? Is E still authorized or not?

cc @arjunhassard @derekpierre @KPrasch

Reply:
Interesting scenario. Given that `E` is used by both `A` and `B`, what scenario could yield that? 🤔 . An encryption service of some sort? But it would be using the same signer for both `A` and `B`? Maybe the user encrypting for themselves on two different apps (`A` and `B`) using the same ritual...?

At first blush, it seems that `E` should still be authorized for `B`. It's unclear the relationship between `A` and `B`, and feels like without that knowledge the safest philosophy is that `A`'s actions should not affect B's.

Did you have existing thoughts here?

Reply:
'B authorizes encryptor E' - that will fail with the current code, encryptor can be authorized only by one auth admin

Reply:
Right, if B can't authorize E becasue it's already authorized by A, then the problem disappears. We need to think carefully if this is acceptable, but if so, then I don't have further comments.

Reply:
Would the following scenario be a possibility? Two apps, `A` and `B`, each being authAdmins for the same ritual (use as part of using the same broader service) and wanting to control their own set of encryptors. The encryptor could be the user and the user utilizes the same wallet across the two apps?

Reply:
could you elaborate a bit more
so we have two auth admins? and same ritual? each of them can have own set of encryptors within paid subscription

Reply:
Yes. Effectively, two auth admins for the same ritual. e.g. Ceramic pays for a ritual, `r`, for their developers, and then assigns two separate developers/application owners, `A` and `B`, as auth admins as part of their respective applications.

Both applications allow their individual users to be encryptors.  User, `U`, uses both apps and wants to use their EOA wallet as an encryptor...?